packages/i18n-calypso/_test/index.js

```diff
@@ -0,0 +1,366 @@
+ * @return {string}        html with react attributes removed
+ */
+function stripReactAttributes( string ) {
+	return string.replace( /\sdata\-(reactid|react\-checksum)\=\"[^\"]+\"/g, '' );
+}
+
+describe( 'I18n', function() {
+	beforeEach( function() {
+		i18n.setLocale( data.locale );
+	} );
+
+	afterEach( function() {
+		i18n.configure(); // ensure everything is reset
+	} );
+
+	describe( 'setLocale()', function() {
+		describe( 'adding a new locale source from the same language', function () {
+			beforeEach( function() {
+				i18n.setLocale( {
+					'': data.locale[''],
+					'test1': [
+						'translation1-1'
+					],
+					'test2': [
+						'translation2'
+					],
+					'new translation': [
+						'Neue Übersetzung'
+					]
+				} );
+			} );
+
+			it( 'should make the new translations available', function () {
+				assert.equal( 'Neue Übersetzung', translate( 'new translation' ) );
+			} );
+			it( 'should keep the original translations available as well', function () {
+				assert.equal( 'Aktivieren', translate( 'Activate' ) );
+			} );
+			it( 'should replace existing translations with the new version', function () {
+				assert.equal( 'translation1-1', translate( 'test1' ) );
+				assert.equal( 'translation2', translate( 'test2' ) );
+			} );
+		} );
+
+		describe( 'adding a new locale source from a different language', function () {
+			beforeEach( function() {
+				i18n.setLocale( {
+					'': Object.assign( {}, data.locale[''], {
+						localeSlug: 'fr',
+						'Plural-Forms': 'nplurals=2; plural=n > 1;'
+					} ),
+					'test1': [
+						'traduction1'
+					],
+					'test2': [
+						'traduction2'
+					],
+					'new translation': [
+						'nouvelle traduction'
+					]
+				} );
+			} );
+
+			it( 'should make replace previous locale translations', function () {
+				assert.notEqual( 'translation1', translate( 'test1' ) );
+				assert.equal( 'traduction1', translate( 'test1' ) );
+			} );
+			it( 'should make old translations unavailable', function () {
+				assert.equal( 'Activate', translate( 'Activate' ) );
+			} );
+			it( 'should make new translations available', function () {
+				assert.equal( 'nouvelle traduction', translate( 'new translation' ) );
+			} );
+		} );
+	} );
+
+	describe( 'translate()', function() {
+		describe( 'passing a string', function() {
+			it( 'should find a simple translation', function() {
+				assert.equal( 'translation1', translate( 'test1' ) );
+			} );
+			it( 'should fall back to original string if translation is missing', function() {
+				assert.equal( 'test2', translate( 'test2' ) );
+			} );
+			it( 'should fall back to original if translation isn\'t even null in locale file', function() {
+				assert.equal( 'nonexisting-string', translate( 'nonexisting-string' ) );
+			} );
+		} );
+
+		describe( 'translate with context', function() {
+			it( 'should find a string with context', function() {
+				assert.equal( 'translation3', translate( {
+					original: 'test3',
```

Comment:
:information_source: This string doesn't exist elsewhere without a **context** (or with a different one). You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.